File: konserkita_api/app/Models/Event.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Event extends Model
{
    use HasFactory;

    protected $fillable = [
        'organizer_id',
        'category_id',
        'title',
        'title_en',
        'slug',
        'description',
        'description_en',
        'date',
        'time',
        'location',
        'map_link',
        'banner_image',
        'status',
        'is_numbered_seating',
    ];

    public function organizer(): BelongsTo
    {
        return $this->belongsTo(Organizer::class);
    }

    public function category(): BelongsTo
    {
        return $this->belongsTo(EventCategory::class, 'category_id');
    }

    public function ticketTypes(): HasMany
    {
        return $this->hasMany(TicketType::class);
    }

    public function seatMap(): \Illuminate\Database\Eloquent\Relations\HasOne
    {
        return $this->hasOne(EventSeatMap::class);
    }

    public function reviews(): HasMany
    {
        return $this->hasMany(Review::class);
    }

    protected function bannerImage(): \Illuminate\Database\Eloquent\Casts\Attribute
    {
        return \Illuminate\Database\Eloquent\Casts\Attribute::make(
            get: fn ($value) => $value ? $value : url('default_img.png'),
        );
    }
}

File: konserkita_mobile/lib/models/event.dart
import 'ticket_type.dart';
import 'event_category.dart';

class Event {
  final int id;
  final String title;
  final String? titleEn;
  final String slug;
  final String description;
  final String? descriptionEn;
  final String date;
  final String time;
  final String location;
  final String? bannerImage;
  final String status;
  final bool isNumberedSeating;
  final List<TicketType> ticketTypes;
  final EventCategory? category;

  Event({
    required this.id,
    required this.title,
    this.titleEn,
    required this.slug,
    required this.description,
    this.descriptionEn,
    required this.date,
    required this.time,
    required this.location,
    this.bannerImage,
    required this.status,
    this.isNumberedSeating = false,
    required this.ticketTypes,
    this.category,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    var list = json['ticket_types'] as List? ?? [];
    List<TicketType> ticketTypesList = list.map((i) => TicketType.fromJson(i)).toList();

    return Event(
      id: json['id'],
      title: json['title'],
      titleEn: json['title_en'],
      slug: json['slug'],
      description: json['description'],
      descriptionEn: json['description_en'],
      date: json['date'],
      time: json['time'],
      location: json['location'],
      bannerImage: json['banner_image'],
      status: json['status'],
      isNumberedSeating: json['is_numbered_seating'] == 1 || json['is_numbered_seating'] == true,
      ticketTypes: ticketTypesList,
      category: json['category'] != null ? EventCategory.fromJson(json['category']) : null,
    );
  }
}
